File: Java/Arrays/problem_283. Move Zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int pointer1 = 0;
        int count_zeroes = 0;
        for (int pointer2 = 0; pointer2 < nums.length; pointer2++) {
            if (nums[pointer2] != 0) {
                nums[pointer1++] = nums[pointer2];
            }
            if (nums[pointer2] == 0) {
                count_zeroes++;
            }
        }
        for (int i = nums.length - count_zeroes; i < nums.length; i++) {
            nums[i] = 0;
        }

    }
}

File: Python/Arrays/problem_283. Move Zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        pointer1 = 0
        for pointer2 in range(len(nums)):
            if(nums[pointer2] != 0):
                temp = nums[pointer1]
                nums[pointer1] = nums[pointer2]
                nums[pointer2] = temp
                pointer1 += 1
